src/_guides/environment-declarations.md

```diff
@@ -0,0 +1,187 @@
+
+{{site.alert.note}}
+  The environment declaration constructors are only guaranteed
+  to work when invoked as `const`.
+  Most compilers must be able to evaluate their value at compile time.
+{{site.alert.end}}
+
+Each of the `fromEnvironment` constructors require the
+name or key of the environment declaration key-value pair,
+as well as an optional `defaultValue` override
+that is used when a declaration wasn't defined.
+
+For example, if you only want to log messages only
+when the environment declaration `DEBUG` is set to `true`:
+
+<?code-excerpt "misc/lib/development/environment_declarations.dart (debug-log)"?>
+```dart
+void log(String message) {
+  // Log the debug message if the environment declaration 'DEBUG' is `true`.
+  // If there was no value specified, do not log.
+  if (const bool.fromEnvironment('DEBUG', defaultValue: false)) {
+    print('Debug: $message');
+  }
+}
+```
+
+In this snippet, if `DEBUG` is set to `false`
+during compilation or not specified at all,
+production compilers can completely remove the condition and its body.
+
+The `fromEnvironment` constructors always
+have a default value when the declaration isn't specified.
+Therefore, if you need to specifically check if
+an environment declaration has been specified,
+use the [`bool.hasEnvironment`][bool-has] constructor:
+
+<?code-excerpt "misc/lib/development/environment_declarations.dart (has-debug)"?>
+```dart
+if (const bool.hasEnvironment('DEBUG')) {
+  print('Debug behavior was configured!');
+}
+```
+
+[string-from]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-core/String/String.fromEnvironment.html
+[int-from]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-core/int/int.fromEnvironment.html
+[bool-from]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-core/bool/bool.fromEnvironment.html
+[bool-has]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-core/bool/bool.hasEnvironment.html
+
+## Specifying environment declarations
+
+{{site.alert.warning}}
+  Dart tools and compilers currently do not
+  consistently handle environment declarations
+  with comma-separated values.
+  For more details,
+  see [SDK issue 44995][].
+{{site.alert.end}}
+
+[SDK issue 44995]: https://github.com/dart-lang/sdk/issues/4499
+
+### Dart CLI
+
+Both `dart run` and the `dart compile` subcommands accept
+the `-D` or `--define` options
+to specify environment declaration values.
+
+```terminal
+$ dart run --define=DEBUG=true -DFLAVOR=free main.dart
+$ dart compile exe --define=DEBUG=true -DFLAVOR=free main.dart
+$ dart compile js --define=DEBUG=true -DFLAVOR=free main.dart
+$ dart compile aot-snapshot --define=DEBUG=true -DFLAVOR=free main.dart
+$ dart compile jit-snapshot --define=DEBUG=true -DFLAVOR=free main.dart
+$ dart compile kernel --define=DEBUG=true -DFLAVOR=free main.dart
+```
+
+To specify multiple declarations, 
+you can use multiple define options:
+
+```terminal
+$ dart run --define=DEBUG=true -DFLAVOR=free
+```
```

Comment:
This last section feels spurious after having six lines above showing the same thing already.